### PR TITLE
Expire payments before returning them

### PR DESCRIPTION
**Pull request by danielgranhao**

### CODE REVIEW

File: eel/src/event_handler.rs [hunk 1]
```diff
@@ -60,7 +59,28 @@ impl EventHandler for LipaEventHandler {
...
+                    let payment = match payment.should_be_set_expired() {
+                        false => payment,
+                        true => {
+                            match payment_store
+                                .new_payment_state(&payment.hash, PaymentState::InvoiceExpired)
+                            {
+                                Ok(()) => match payment_store.get_payment(&payment.hash) {
+                                    Ok(p) => p,
+                                    Err(e) => {
+                                        error!("Failed find payment with hash {} when previous attempts have succeeded: {}", payment.hash, e);
```
Comment (gcomte) [inline eel/src/event_handler.rs:72]: Here you handle the bad case, but in the _previous_ attempt that you reference to, the bad case is actually being ignored. ( `if let Ok(payment)` )

Comment (danielgranhao) [reply eel/src/event_handler.rs]: This was removed while addressing another comment. But that's right, at the moment, we claim htlcs even if there's no associated `Payment` in the db. Should we change that? Most likely, it's an htlc from a spontaneous payment. I propose that if we want to change that, we do so in another PR.

File: eel/src/event_handler.rs [hunk 2]
```diff
@@ -60,7 +59,28 @@ impl EventHandler for LipaEventHandler {
 
                 let payment_store = self.payment_store.lock().unwrap();
 
-                if let Ok(payment) = payment_store.get_payment(&payment_hash.0) {
+                if let Ok(payment) = payment_store.get_payment(&payment_hash.0.to_hex()) {
```
[the remark this answers is not in the document]
Comment (danielgranhao) [reply eel/src/event_handler.rs:62]: Addressed in [ab630e3](https://github.com/getlipa/lipa-lightning-lib/pull/305/commits/ab630e398e7b54cd8a489745284244be09bed89c)

File: eel/src/event_handler.rs (hunk 1 above)
Comment (gcomte) [reply eel/src/event_handler.rs]: Oh okay. I think it's fine.

File: eel/src/payment_store.rs [hunk 3]
```diff
@@ -81,6 +80,26 @@ pub struct Payment {
...
+                PaymentType::Sending => {
+                    if self.payment_state == PaymentState::Failed {
+                        return true;
+                    }
+                }
+            }
+        }
+        false
+    }
+}
```
[the remark this answers is not in the document]
Comment (danielgranhao) [reply eel/src/payment_store.rs:101]: Fair, but IMO, that makes the code less readable.

[the remark this answers is not in the document]
Comment (gcomte) [reply eel/src/payment_store.rs:101]: ok

**Review by gcomte — APPROVED** [review]

**Pull request merged**